src/main/java/uk/gov/hmcts/cmc/claimstore/services/ClaimantResponseService.java

```diff
@@ -62,23 +63,25 @@ public ClaimantResponseService(
     public void save(
         String externalId,
         String claimantId,
-        ClaimantResponse response,
+        ClaimantResponse claimantResponse,
         String authorization
     ) {
         Claim claim = claimService.getClaimByExternalId(externalId, authorization);
         claimantResponseRule.assertCanBeRequested(claim, claimantId);
 
-        Claim updatedClaim = caseRepository.saveClaimantResponse(claim, response, authorization);
+        Claim updatedClaim = caseRepository.saveClaimantResponse(claim, claimantResponse, authorization);
         claimantResponseRule.isValid(updatedClaim);
-        formaliseResponseAcceptance(response, updatedClaim, authorization);
-        if (isRejectPartAdmitNoMediation(response, updatedClaim)) {
+        formaliseResponseAcceptance(claimantResponse, updatedClaim, authorization);
+        if (isRejectPartAdmitNoMediation(claimantResponse, updatedClaim)) {
             updateDirectionsQuestionnaireDeadline(updatedClaim, authorization);
         }
-        if (!isReferredToJudge(response)) {
+        Response response = claim.getResponse().orElseThrow(IllegalArgumentException::new);
```

Comment:
should we do this higher up? - e.g. after `claimantResponseRule.assertCanBeRequested(claim, claimantId);`

Reply:
assertCanBeRequested is doing this check inside as a validation check bit differently but we wouldn't be using response until few lines later and checkstyle is likely to complain. Check `VariableDeclarationUsageDistance`